src/app/core/guards/auth.guard.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, CanLoad, Router, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate, CanLoad {
+
+  constructor(private router: Router) {}
+
+  canActivate(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.checkToken();
+  }
+
+  canLoad(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.checkToken();
+  }
+
+  public checkToken() {
+    if (localStorage.getItem('authToken')) {
+      return true;
+    }
+    this.router.navigateByUrl('/login');
```

Comment:
return this.router.parseUrl('login')

that will return UrlTree, as you mentioned in return type of your canActivate and canLoad

Reply:
Done